api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -265,6 +265,22 @@ public boolean equals(Object other) {
     return Arrays.equals(fields, that.fields);
   }
 
+  public boolean equalOrFinerThan(PartitionSpec that) {
+    if (this.fields.length < that.fields.length) {
+      return false;
+    }
+
+    for (int i = 0; i < that.fields.length; i++) {
+      if (this.fields[i].sourceId() != that.fields[i].sourceId()) {
+        return false;
+      }
```

Comment:
nit: newline after this. See: https://iceberg.apache.org/contribute/#block-spacing